<a href="https://colab.research.google.com/github/TheKoray18/Diabetes-Classification-with-Neural-Network-/blob/main/Diabetes.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive

drive.mount('/content/drive/')

Mounted at /content/drive/


- Datamızı Çağırıyoruz

In [48]:
import pandas as pd 

data=pd.read_csv('/content/drive/My Drive/Colab Notebooks/pima-indians-diabetes.csv')

- Datamızın ilk 5 satırına bakıyoruz

In [49]:
data.head()

6  148  72  35    0  33.6  0.627  50  1
0  1   85  66  29    0  26.6  0.351  31  0
1  8  183  64   0    0  23.3  0.672  32  1
2  1   89  66  23   94  28.1  0.167  21  0
3  0  137  40  35  168  43.1  2.288  33  1
4  5  116  74   0    0  25.6  0.201  30  0

- Datamızın bilgilerine bakıyoruz


In [50]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 767 entries, 0 to 766
Data columns (total 9 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   6       767 non-null    int64  
 1   148     767 non-null    int64  
 2   72      767 non-null    int64  
 3   35      767 non-null    int64  
 4   0       767 non-null    int64  
 5   33.6    767 non-null    float64
 6   0.627   767 non-null    float64
 7   50      767 non-null    int64  
 8   1       767 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


- Datamızda ki Missing Value değerlerine bakıyoruz

In [51]:
data.isnull()

6    148     72     35      0   33.6  0.627     50      1
0    False  False  False  False  False  False  False  False  False
1    False  False  False  False  False  False  False  False  False
2    False  False  False  False  False  False  False  False  False
3    False  False  False  False  False  False  False  False  False
4    False  False  False  False  False  False  False  False  False
..     ...    ...    ...    ...    ...    ...    ...    ...    ...
762  False  False  False  False  False  False  False  False  False
763  False  False  False  False  False  False  False  False  False
764  False  False  False  False  False  False  False  False  False
765  False  False  False  False  False  False  False  False  False
766  False  False  False  False  False  False  False  False  False

[767 rows x 9 columns]

In [52]:
data.isnull().any()

6        False
148      False
72       False
35       False
0        False
33.6     False
0.627    False
50       False
1        False
dtype: bool

- Datamıza baktığımızda columns names olmadığını görmüştük
- Bu yüzden her sütun(column) için bir uygun isim atıyoruz

In [54]:
data.columns=["pregnancies","glucose","blood_pressure","skin_thickness","insulin",
                                                      "bmi","Diabetes_pedigre","age","outcome"]

In [55]:
data

pregnancies  glucose  blood_pressure  ...  Diabetes_pedigre  age  outcome
0              1       85              66  ...             0.351   31        0
1              8      183              64  ...             0.672   32        1
2              1       89              66  ...             0.167   21        0
3              0      137              40  ...             2.288   33        1
4              5      116              74  ...             0.201   30        0
..           ...      ...             ...  ...               ...  ...      ...
762           10      101              76  ...             0.171   63        0
763            2      122              70  ...             0.340   27        0
764            5      121              72  ...             0.245   30        0
765            1      126              60  ...             0.349   47        1
766            1       93              70  ...             0.315   23        0

[767 rows x 9 columns]

- Column Names atadıktan sonra data bilgilerimize yeniden bakıyoruz


In [58]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 767 entries, 0 to 766
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   pregnancies       767 non-null    int64  
 1   glucose           767 non-null    int64  
 2   blood_pressure    767 non-null    int64  
 3   skin_thickness    767 non-null    int64  
 4   insulin           767 non-null    int64  
 5   bmi               767 non-null    float64
 6   Diabetes_pedigre  767 non-null    float64
 7   age               767 non-null    int64  
 8   outcome           767 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


- Datamızda "0" olarak eksik veriler var.Bizim bunu tabiki düzeltmemiz,doldurmamız gerekiyor

- İlk olarak 0 olan eksik veriyi NaN olarak düzelteceğiz

In [59]:
import numpy as np

data.glucose.replace(0,np.nan,inplace=True)
data.blood_pressure.replace(0,np.nan,inplace=True)
data.skin_thickness.replace(0,np.nan,inplace=True)
data.insulin.replace(0,np.nan,inplace=True)
data.bmi.replace(0,np.nan,inplace=True)
data.Diabetes_pedigre.replace(0,np.nan,inplace=True)
data.age.replace(0,np.nan,inplace=True)

- Aşağıda göreceğimiz gibi 0 olan eksik veriler NaN olmuş durumda

In [60]:
data.head()

pregnancies  glucose  blood_pressure  ...  Diabetes_pedigre  age  outcome
0            1     85.0            66.0  ...             0.351   31        0
1            8    183.0            64.0  ...             0.672   32        1
2            1     89.0            66.0  ...             0.167   21        0
3            0    137.0            40.0  ...             2.288   33        1
4            5    116.0            74.0  ...             0.201   30        0

[5 rows x 9 columns]

- NaN olarak değiştirdiğimiz eksik verilerimizi kendi sütunundaki değerlerin ortalamasını(mean) alarak dolduracağız

In [61]:
data.fillna(data.mean(),inplace=True)

- Aşağıda göreceğimiz gibi eksik NaN olan eksik verilerimizi bir değer ile doldurmuş durumdayız.

- Doldurmuş olduğumuz değer,o sütundaki değerlerin ortalamasına eşit olan değerdi

In [62]:
data.head()

pregnancies  glucose  blood_pressure  ...  Diabetes_pedigre  age  outcome
0            1     85.0            66.0  ...             0.351   31        0
1            8    183.0            64.0  ...             0.672   32        1
2            1     89.0            66.0  ...             0.167   21        0
3            0    137.0            40.0  ...             2.288   33        1
4            5    116.0            74.0  ...             0.201   30        0

[5 rows x 9 columns]

- Bağımsız değişkenimiz olan outcome(diabet sonuçları) datamızdan alıyoruz

- Biz outcome sınıflandırması yapacağımız için datamızdan outcome aldık

- Yani hastaların diabet olup olmadığının sınıflandırılmasını yapacağız

In [63]:
outcome=data.iloc[:,-1]

In [64]:
outcome

0      0
1      1
2      0
3      1
4      0
      ..
762    0
763    0
764    0
765    1
766    0
Name: outcome, Length: 767, dtype: int64

- Datamızdan outcome aldığımıza göre şimdi onu bağımlı değişkenimiz olan data'mızdan siliyoruz

In [65]:
data.drop(['outcome'],axis=1,inplace=True)

- Aşağıda göreceğimiz gibi outcome sütunu artık datamızda yok

In [66]:
data

pregnancies  glucose  blood_pressure  ...   bmi  Diabetes_pedigre  age
0              1     85.0            66.0  ...  26.6             0.351   31
1              8    183.0            64.0  ...  23.3             0.672   32
2              1     89.0            66.0  ...  28.1             0.167   21
3              0    137.0            40.0  ...  43.1             2.288   33
4              5    116.0            74.0  ...  25.6             0.201   30
..           ...      ...             ...  ...   ...               ...  ...
762           10    101.0            76.0  ...  32.9             0.171   63
763            2    122.0            70.0  ...  36.8             0.340   27
764            5    121.0            72.0  ...  26.2             0.245   30
765            1    126.0            60.0  ...  30.1             0.349   47
766            1     93.0            70.0  ...  30.4             0.315   23

[767 rows x 8 columns]

- Datamızı ölçeklendireceğiz.Standarlaştırma yapacağız.
- Makinemiz daha iyi bir öğrenme gerçekleştirme yapması için bunu yapıyoruz
- Modelimizin daha iyi performans gerçekleştirmesi için


In [67]:
from sklearn.preprocessing import StandardScaler
s_sca=StandardScaler()

data=s_sca.fit_transform(data)

In [68]:
data

array([[-0.84372629, -1.20482941, -0.52956011, ..., -0.8517715 ,
        -0.36426474, -0.18894038],
       [ 1.23423997,  2.01661929, -0.69489941, ..., -1.33176959,
         0.60470064, -0.1037951 ],
       [-0.84372629, -1.0733417 , -0.52956011, ..., -0.63359055,
        -0.91968415, -1.0403932 ],
       ...,
       [ 0.343683  , -0.02144009, -0.03354219, ..., -0.90995308,
        -0.68423462, -0.27408566],
       [-0.84372629,  0.14291954, -1.02557802, ..., -0.34268262,
        -0.37030191,  1.17338414],
       [-0.84372629, -0.941854  , -0.1988815 , ..., -0.29904643,
        -0.47293375, -0.87010264]])

- Datamızı Train ve Test Olarak Ayırıyoruz

In [91]:
from sklearn.model_selection import train_test_split

x_train,x_test,y_train,y_test=train_test_split(data,outcome,test_size=0.33,random_state=0)

- Train ve Test Datalarımızın Boyutlarına bakıyoruz

In [92]:
print("x_train="+str(x_train.shape))
print("x_test="+str(x_test.shape))
print("y_train="+str(y_train.shape))
print("y_test="+str(y_test.shape))

x_train=(513, 8)
x_test=(254, 8)
y_train=(513,)
y_test=(254,)


In [93]:
x_train

array([[ 2.12479694,  0.53738265,  0.13179711, ...,  0.53004118,
         0.25756351,  1.42881999],
       [ 0.04683068,  0.96471768,  1.45451155, ..., -0.4008642 ,
        -0.53632401,  0.23678603],
       [ 1.82794462,  0.24153532, -0.86023872, ...,  1.27185641,
        -0.0925922 ,  0.4070766 ],
       ...,
       [ 0.93738765, -0.25154357, -0.69489941, ..., -0.73540833,
         0.78581567,  0.06649547],
       [ 0.64053533,  0.11004761,  0.29713641, ...,  0.19549706,
        -1.058539  ,  1.76940112],
       [-0.54687397,  0.24153532,  0.13179711, ...,  0.10822468,
         0.36019536, -0.69981207]])

- Bizim y_train datamızın boyutu (513,).Yani 513 satırlı(row) bir datadır

- Ama biz y_train yazdırdığımızda 767 tane satır sayılarından oluşan dağınık bir şekilde 513 satır oluşturuyor

- Bu durum bizim modelimizi test ederken işimizi zorlaştıracak

- İlk olarak reset_index() ile bir index sütunu oluşturuyoruz.Bu index sütunumuzun değerleri ise bizim sıralı olmayan dağınık şekilde sıralanan satır sayılarımız oluyor

- Aynı zaman da 0'dan 513'e kadar olan satır sayıları oluşturuyor

- Aşağıda göreceğiz gibi 

In [94]:
y_train.reset_index()

index  outcome
0      613        1
1      159        0
2      711        1
3      745        1
4      447        1
..     ...      ...
508    707        1
509    192        1
510    629        1
511    559        1
512    684        0

[513 rows x 2 columns]

- Bizim 0'dan 513'e kadar olan satır sayılarımız , 717'ye kadar olan dağınık olarak yani sıralı olmayan şekilde numaralandırılmış index sütunumuz ve outcome sütunumuz vardı

- Şimdi biz drop=True diyerek sıralı olmayan,dağınık şekilde numaralandırılmış index sütununu datamızdan siliyoruz

- Şimdi bizim y_train datamız da outcome değerlerimiz ve bu değerlere karşılık gelen satır sayılarımız oluyor

- Satır sayılarımız 0'dan 513'e kadar olan sayılarımız

In [95]:
y_train=y_train.reset_index(drop=True)

In [96]:
y_train

0      1
1      0
2      1
3      1
4      1
      ..
508    1
509    1
510    1
511    1
512    0
Name: outcome, Length: 513, dtype: int64

- y_train datamız çin yaptığımız index işlemlerimizin aynısını y_test için de gerçekleştiriyoruz

In [97]:
y_test.reset_index()

index  outcome
0      661        1
1      122        0
2      113        1
3       14        1
4      529        0
..     ...      ...
249    710        0
250    531        0
251    666        1
252    252        0
253    293        0

[254 rows x 2 columns]

In [98]:
y_test=y_test.reset_index(drop=True)

In [99]:
y_test

0      1
1      0
2      1
3      1
4      0
      ..
249    0
250    0
251    1
252    0
253    0
Name: outcome, Length: 254, dtype: int64

In [100]:
x_test

array([[ 1.23423997,  1.49066849,  2.77722598, ...,  0.74822213,
        -0.92572132,  0.83280301],
       [ 0.343683  ,  0.34015109,  0.62781502, ..., -0.82268071,
        -0.86233106,  3.04658036],
       [ 0.93738765,  1.26056501, -1.52159594, ..., -0.28450103,
         0.35113961,  0.49222188],
       ...,
       [ 1.82794462, -0.35015934, -0.1988815 , ..., -0.72086293,
        -0.99816733,  0.57736716],
       [-1.14057861, -1.17195748, -0.3642208 , ...,  0.48640499,
        -0.7053647 , -0.69981207],
       [-1.14057861,  1.29343693, -1.85227454, ..., -1.53540514,
        -0.65706736,  2.70599923]])

- Logistic Regression kullanarak datamızı x_train ve y_train ile train(fit) ediyoruz


In [ ]:
from sklearn.linear_model import LogisticRegression 

log_reg=LogisticRegression(random_state=0)

log_reg.fit(x_train,y_train)

- x_test datamızı kullanarak predict metodu ile modelimizi test ediyoruz

- Yani prediction değerleri elde ederiz

In [102]:
y_pred=log_reg.predict(x_test)

In [103]:
y_pred

array([1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0,
       0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0,
       0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 1,
       0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0,
       1, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0,
       1, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 1, 0,
       1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0,
       1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0,
       1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0,
       0, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0,
       0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0])

- Confusion Matrix oluşturuyoruz

In [109]:
from sklearn.metrics import confusion_matrix

cm=confusion_matrix(y_test,y_pred)

print("Confusion Matrix:\n",cm)

Confusion Matrix:
 [[152  24]
 [ 36  42]]


- Prediction(tahminler), y_test verilerimiz ile test ediyoruz.

- For döngüsü ile İlk 25 veriyi deniyoruz 

- True_pre ve False_pre adında 2 tane list oluşturuyorum

- True_pre içerisine doğru prediction yapılan verilerin satır sayısını ekliyoruz

- False_pre içerisine de yanlış prediction yapılan verilerin satır sayılarını ekliyoruz 

In [116]:
True_pre=[]
False_pre=[]

for i in range(25):
  print("y_test= {} - Prediction= {}".format(y_test[i],y_pred[i]))
  if y_test[i] == y_pred[i]:
    print("True Prediction\n")
    True_pre.append(i)
  else:
    print("False Prediction\n")
    False_pre.append(i)

y_test= 1 - Prediction= 1
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 1 - Prediction= 1
True Prediction

y_test= 1 - Prediction= 0
False Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 1 - Prediction= 1
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 1 - Prediction= 1
True Prediction

y_test= 1 - Prediction= 1
True Prediction

y_test= 1 - Prediction= 0
False Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 0 - Prediction= 0
True Prediction

y_test= 0

- Aşağıda göreceğimiz gibi hangi satırlarda ki değerler de True prediction yapmış hangi satırlarda False Prediction yapmış görüyoruz

In [122]:
print("True Prediction= \n{}\n".format(True_pre))
print("False Prediction= \n{}".format(False_pre))

True Prediction= 
[0, 1, 2, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 20, 21, 22, 23]

False Prediction= 
[3, 19, 24]


- İlk 25 veride yapılan True prediction ve False Prediction başarı oranlarını hesapladım

- t_num ve f_num tanımladım.

- t_num = Yapılan Doğru tahminlerin sayısı

- f_num= Yapılan Yanlış tahminlerin sayısı

- Numpy kullanarak yüzdelik hesabı yaptım

- True Prediction için true_rate , False Prediction için false_rate hesapladım

In [123]:
t_num=len(True_pre)
f_num=len(False_pre)

In [130]:
true_rate=np.dot((np.divide(t_num,25)),100)

false_rate=np.dot((np.divide(f_num,25)),100)

- True Prediction oranını ve False Prediction oranını aşağıda göreceksiniz


In [133]:
print("True Prediction Rate = {}\n".format(true_rate))
print("False Prediction Rate = {}".format(false_rate))

True Prediction Rate = 88.0

False Prediction Rate = 12.0


- Diabet hasta sınıflandırılmasını Sinir Ağı kullanarak yapacağız.

- Tensorflow kullancağız

In [134]:
import tensorflow as tf 

- 4 hidden layer,katmanları 256,128 olan bir sinir ağı oluşturduk

- Overfitting'i engellemek için 2 kez Droput uyguladık.Biri 0.25 oranında biri ise 0.5 oranında 

- Activasyon fonksiyonu olarak 'relu' kullandık

- Son hidden layer'da aktivasyon fonskyionu olarak 'Sigmoid' kullandık.Bunun sebebi,Biz 0-1'lerden oluşan bir sınıflandırma yapıyoruz.Sigmoid de bize 0 ile 1 arasında probabilitik değerler döndürür

- Optimizer olarak 'adam' kullandık

- loss fonksiyonu olarak 'Binary_crossentropy' kullandık.Biz 0-1 sınıflandırma yapacağız.Yanı binary bir sınıflandırma yapacagız.O yüzden binary_crossentropy kullandık

- Metrics ölçüm parametmerimiz 'accuracy'

- Eğitim sayısını(epochs) 50 olarak belirledik

In [149]:
def Train():

  model=tf.keras.models.Sequential([tf.keras.layers.Dense(units=256,activation='relu',input_dim=8),
                                                          tf.keras.layers.Dense(units=256,activation='relu'),
                                                          tf.keras.layers.Dense(units=128,activation='relu'),
                                                          tf.keras.layers.Dropout(0.25),
                                                          tf.keras.layers.Dense(units=128,activation='relu'),
                                                          tf.keras.layers.Dropout(0.5),
                                                          tf.keras.layers.Dense(units=1,activation='sigmoid')
                                                          ])
  model.compile(optimizer='adam',loss='binary_crossentropy',metrics=['accuracy'])
  
  history=model.fit(x_train,y_train,epochs=50,validation_data=(x_test,y_test),verbose=1)

  test_loss,test_acc=model.evaluate(x_test,y_test,verbose=2)

  cache={"test_loss":test_loss,"test_acc":test_acc}

  history={"model":model}
    
  return cache,history

In [150]:
cache,history=Train()

Epoch 1/50
17/17 [==============================] - 0s 14ms/step - loss: 0.6099 - accuracy: 0.6842 - val_loss: 0.4659 - val_accuracy: 0.7677
Epoch 2/50
17/17 [==============================] - 0s 5ms/step - loss: 0.5000 - accuracy: 0.7583 - val_loss: 0.4502 - val_accuracy: 0.7559
Epoch 3/50
17/17 [==============================] - 0s 5ms/step - loss: 0.4675 - accuracy: 0.7739 - val_loss: 0.4441 - val_accuracy: 0.7756
Epoch 4/50
17/17 [==============================] - 0s 5ms/step - loss: 0.4465 - accuracy: 0.7953 - val_loss: 0.4480 - val_accuracy: 0.7638
Epoch 5/50
17/17 [==============================] - 0s 5ms/step - loss: 0.4315 - accuracy: 0.8070 - val_loss: 0.4579 - val_accuracy: 0.7638
Epoch 6/50
17/17 [==============================] - 0s 6ms/step - loss: 0.4374 - accuracy: 0.8109 - val_loss: 0.4616 - val_accuracy: 0.7638
Epoch 7/50
17/17 [==============================] - 0s 5ms/step - loss: 0.4126 - accuracy: 0.8031 - val_loss: 0.4666 - val_accuracy: 0.7756
Epoch 8/50
17/17 [=

- Test Loss ve Test Acc değerlerimize bakıyoruz

In [151]:
print("test_loss="+str(cache['test_loss']))
print("test_acc="+str(cache['test_acc']))

test_loss=1.3445571660995483
test_acc=0.7716535329818726


- x_test datamızı kullanarak predicton değerleri oluşturuyoruz

In [152]:
model=history['model']

y_pred_nn=model.predict(x_test)

In [172]:
y_pred_nn

array([[ True],
       [False],
       [ True],
       [False],
       [False],
       [ True],
       [ True],
       [False],
       [False],
       [False],
       [False],
       [False],
       [False],
       [ True],
       [False],
       [False],
       [False],
       [ True],
       [ True],
       [False],
       [False],
       [False],
       [False],
       [False],
       [False],
       [ True],
       [False],
       [ True],
       [False],
       [False],
       [False],
       [False],
       [False],
       [ True],
       [ True],
       [ True],
       [False],
       [False],
       [False],
       [ True],
       [False],
       [ True],
       [False],
       [False],
       [False],
       [False],
       [False],
       [ True],
       [False],
       [False],
       [False],
       [False],
       [False],
       [False],
       [False],
       [False],
       [ True],
       [False],
       [False],
       [False],
       [ True],
       [False],
       [

- Sigmoid fonksiyonu bize 0-1 arasında değerler döndürüyordu

- Bu yüzden y_pred_nn datasında 0-1 arasında olan değerler vardır

- Ama bizim verimizi test etmek için ya 0 ya da 1 olması lazım.Bu yüzden bizde > 0.5 işlemi yaparak ;

    -- 0.5 ' den büyük değerler için 1 

    -- 0.5 ' den küçük değerler için 0 olmasını istedik

- Bize 0-1 değerler döndürmüyor . True -False Boolean değerler döndürüyor

   -- 1 için True 

   -- 0 için False döndürüyor

In [160]:
y_pred_nn = y_pred_nn > 0.5

In [173]:
y_pred_nn

array([[ True],
       [False],
       [ True],
       [False],
       [False],
       [ True],
       [ True],
       [False],
       [False],
       [False],
       [False],
       [False],
       [False],
       [ True],
       [False],
       [False],
       [False],
       [ True],
       [ True],
       [False],
       [False],
       [False],
       [False],
       [False],
       [False],
       [ True],
       [False],
       [ True],
       [False],
       [False],
       [False],
       [False],
       [False],
       [ True],
       [ True],
       [ True],
       [False],
       [False],
       [False],
       [ True],
       [False],
       [ True],
       [False],
       [False],
       [False],
       [False],
       [False],
       [ True],
       [False],
       [False],
       [False],
       [False],
       [False],
       [False],
       [False],
       [False],
       [ True],
       [False],
       [False],
       [False],
       [ True],
       [False],
       [

- İlk 25 değer için y_test ile prediction değerlerimizi karşılaştırıyoruz

In [162]:
for i in range(25):
  print("y_test = {} - prediction = {}".format(y_test[i],y_pred_nn[i]))

y_test = 1 - prediction = [ True]
y_test = 0 - prediction = [False]
y_test = 1 - prediction = [ True]
y_test = 1 - prediction = [False]
y_test = 0 - prediction = [False]
y_test = 0 - prediction = [ True]
y_test = 1 - prediction = [ True]
y_test = 0 - prediction = [False]
y_test = 0 - prediction = [False]
y_test = 0 - prediction = [False]
y_test = 0 - prediction = [False]
y_test = 0 - prediction = [False]
y_test = 0 - prediction = [False]
y_test = 0 - prediction = [ True]
y_test = 0 - prediction = [False]
y_test = 0 - prediction = [False]
y_test = 0 - prediction = [False]
y_test = 1 - prediction = [ True]
y_test = 1 - prediction = [ True]
y_test = 1 - prediction = [False]
y_test = 0 - prediction = [False]
y_test = 0 - prediction = [False]
y_test = 0 - prediction = [False]
y_test = 0 - prediction = [False]
y_test = 0 - prediction = [False]


- Confusion Matrix Oluşturuyoruz

In [163]:
cm_nn=confusion_matrix(y_test,y_pred_nn)

print("NN Confusion Matrix = \n{}".format(cm_nn))

NN Confusion Matrix = 
[[152  24]
 [ 34  44]]


In [146]:
from sklearn.metrics import accuracy_score

accsco=accuracy_score(y_test,y_pred)

- Validation Accuracy Score'a bakıyoruz


In [148]:
print("Val Accuracy Score = {}".format(accsco))

Val Accuracy Score = 0.7637795275590551


- Sinir Ağımız,Logistic Regression algoritmamıza göre Diabetli hasta sınıflandırılmasında daha iyi sonuç veriyor 

- Train accuracy score = 0.97

- Test accuracy score = 0.76. Aralarında ki fark biraz fazladır.Yani bu demek oluyor ki modelimiz ezberleme yapmış.Bunu Dropout veya Regularization kullanarak engelleyebiliriz.